internal/parse/parser.go

```diff
@@ -243,58 +236,65 @@ func (p *Parser) skipName() bool {
 //  - bool == false, err == nil
 //		The construct was not the one we are looking for
 
-// parseIdentifier parses either a name made up only of nameBytes or an
-// asterisk.
-func (p *Parser) parseIdentifier() (string, bool) {
-	if p.skipByte('*') {
-		return "*", true
-	}
+type asteriskFlag int
 
-	idStart := p.pos
+const (
+	allowAsterisk asteriskFlag = iota
+	disallowAsterisk
+)
+
+// parseIdentifier parses a name made up of only nameBytes, or alternatively a
+// single asterisk if `allowAsterisk` is passed. On success it returns the
+// parsed string and true. Otherwise, it returns the empty string and false.
+func (p *Parser) parseIdentifier(asteriskF asteriskFlag) (string, bool) {
+	if asteriskF == allowAsterisk {
```

Comment:
Instead of having a flag, why not just having two functions:
```
func parseAsteriskIdentifier() (string, bool) {
     if p.skipByte(...) {
         return "*", ...
     }
    return parseIdentifier()
}

func parseIdentifier() (string, bool) {
    return
}
```

Up to you, though.

Reply:
Changed. We were trying to work out the most idiomatic way to do this but I think separate functions is nicer than what we had.

Reply:
Changed. We were trying to work out the most idiomatic way to do this but I think separate functions is nicer than what we had.